### PR TITLE
Vuefy library folder details modal 

### PR DESCRIPTION
**Pull request by OlegZharkov**
This is a playground for our rising star front-end developer @davelopez!

## Update
_(by @davelopez)_
Thank you @OlegZharkov for setting up the playground and for all the help! I learned a lot :laughing:

This PR migrates the library folder details modal dialog to a Vue component and adds a couple of tests.
This is a sample of the final result (pretty much the same as before but now I know a bit of vue-fu :rofl:):

Screenshot from 2021-11-13 19-02-59

I have noticed the unsettlingly accurate 'pretty' time in the `created` property coming from `pretty_print_time_interval` in the server. I wonder if this kind of time representation should be actually handled by the client and not by the backend at all :thinking: 

## How to test the changes?
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  - Go to *Shared Data -> Data Libraries*.
  - Select a library and optionally a sub-folder.
  - Click the `Details` button in the top menu bar.

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: client/src/components/Libraries/LibraryFolder/FolderDetails/Details.vue [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            try {
+                const url = `${getAppRoot()}api/libraries/${this.metadata.parent_library_id}`;
+                const response = await axios.get(url);
+
+                this.libraryDetails = buildLibraryDetailsFields(response.data);
+                this.folderDetails = [
+                    { field: "Name", value: this.metadata.folder_name },
+                    { field: "Description", value: this.metadata.folder_description },
+                    { field: "ID", value: this.id },
+                ];
```
Comment (OlegZharkov) [inline client/src/components/Libraries/LibraryFolder/FolderDetails/Details.vue:85]:
```suggestion
                this.folderDetails = Object.keys(this.metadata).map((key) => ({ key, value: this.metadata[key] }));
```

File: lib/galaxy/selenium/navigation.yml [hunk 2]
```diff
@@ -795,6 +795,8 @@ libraries:
       make_private_btn: '#make-private'
       access_dataset_roles: '.access_dataset_roles .multiselect__tag span'
       private_dataset_icon: '.fa-key'
+      open_location_details_btn: '.details-btn'
+      location_details_ok_btn: '.modal .btn-primary'
```
Comment (OlegZharkov) [inline lib/galaxy/selenium/navigation.yml]:
```suggestion
      location_details_ok_btn: '#details-modal .btn-primary'
```

**Comment by dannon** [comment]
> I have noticed the unsettlingly accurate 'pretty' time in the `created` property coming from `pretty_print_time_interval` in the server. I wonder if this kind of time representation should be actually handled by the client and not by the backend at all 🤔

Yep, we should just serve an accurate datetime and rely on client(s) to format it as they wish.

**Comment by OlegZharkov** [comment]
> > I have noticed the unsettlingly accurate 'pretty' time in the `created` property coming from `pretty_print_time_interval` in the server. I wonder if this kind of time representation should be actually handled by the client and not by the backend at all thinking
> 
> Yep, we should just serve an accurate datetime and rely on client(s) to format it as they wish.

I just remembered that I thought a lot with this like a year ago, just let me search that PR ....

**Comment by OlegZharkov** [comment]
> > I have noticed the unsettlingly accurate 'pretty' time in the `created` property coming from `pretty_print_time_interval` in the server. I wonder if this kind of time representation should be actually handled by the client and not by the backend at all thinking
> 
> Yep, we should just serve an accurate datetime and rely on client(s) to format it as they wish.

Fixed it! :)
image


the old version (does it make sense to remove/fix it on backend?):
image

**Comment by dannon** [comment]
> Fixed it! :)

Nice!  Looks good.

**Review by dannon — APPROVED** [review]

**Comment by davelopez** [comment]
Thank you @dannon!

**Comment by OlegZharkov** [comment]
thanks @dannon!

**Pull request merged**